src/Functions/PositionImpl.h

```diff
@@ -202,6 +202,51 @@ struct PositionImpl
         const UInt8 * const end = haystack_data.data() + haystack_data.size();
         const UInt8 * pos = begin;
 
+        /// Fastpath when needle is empty
+        if (needle.empty())
+        {
+            /// When needle is empty and start_pos doesn't exist, always return 1
+            if (start_pos == nullptr)
+            {
+                for (auto & x : res)
+                    x = 1;
+                return;
+            }
+
+            ColumnString::Offset prev_offset = 0;
+            auto rows = haystack_offsets.size();
+            const ColumnConst * start_pos_const = typeid_cast<const ColumnConst *>(&*start_pos);
```

Comment:
Please
- put in a newline after l. 217
- use `size_t` instead of `auto` in l. 217 for consistency with the rest of the file
- and then replace l. 218-221 by:

```cpp
/// When needle is empty and start_pos is constant
if (const ColumnConst * start_pos_const = typeid_cast<const ColumnConst *>(&*start_pos))
{
```

Also, for symmetry, the code in l. 236 could be moved into an `else` branch of l. 219.